plan/column_pruning.go

```diff
@@ -261,6 +262,11 @@ func (p *LogicalJoin) PruneColumns(parentUsedCols []*expression.Column) {
 	rChild := p.children[1].(LogicalPlan)
 	lChild.PruneColumns(leftCols)
 	rChild.PruneColumns(rightCols)
+	if p.JoinType == LeftOuterJoin {
+		p.DefaultValues = make([]types.Datum, p.children[1].Schema().Len())
```

Comment:
DefaultValues = DefaultValues[:p.children[1].Schema().Len()] ?

Reply:
It seems that this will cause panic, i'll look into this tomorrow.

Reply:
This is caused by `USING` clause in join which may make the schema bigger than before.